### DISCUSSION TITLE
`json` and `script/en` directories

### DISCUSSION THREAD

**Discussion by andrey-helldar** [Laravel-Lang/common#44] [Q&A] [answered]
Previously, the `script` folder contained the source files of localizations in English, as well as the todo.md generation script.

Recently, this script has been revised and moved to another folder.

Also, translation keys for json have been added to the project.

Now the project structure is as follows:

- `script/en/` - stores php and json source files
- `json/` - stores json translation files
- `src/` - stores php translation files

I suggest changing the structure to this:

- `source/` - stores php and json source files
- `locales/{locale}` - stores php and json files with translations.

For example:

- `source/en.json`
- `source/auth.php`
- `source/pagination.php`
- `source/passwords.php`
- `source/validation.php`
- `source/validation-inline.php`
- `locales/fr/fr.json`
- `locales/fr/auth.php`
- `locales/fr/pagination.php`
- `locales/fr/passwords.php`
- `locales/fr/validation.php`
- `locales/fr/validation-inline.php`

Of course, this is a very large change in the package logic, so it can be released in version 9.

At the same time, it is not difficult to change the local project scripts and translation managers - just need to specify a new path.

@Laravel-Lang/laravel-lang, what do you think?

**Comment by caouecs** [comment]
the idea is good, and yes, it's for version 9

**Comment by mohamedsabil83** [comment, marked as answer]
I agree with this as it will be more organized. Also, good to be version 9.

**Comment by luisprmat** [comment]
I agree

**Comment by andrey-helldar** [comment]
PR: https://github.com/Laravel-Lang/lang/pull/1607
